analyzers/src/SonarAnalyzer.Common/Rules/Utilities/UtilityAnalyzerBase.cs

```diff
@@ -34,13 +35,19 @@ public abstract class UtilityAnalyzerBase : SonarDiagnosticAnalyzer
         private const string ProjectOutFolderPathFileName = "ProjectOutFolderPath.txt";
 
         protected static readonly ISet<string> FileExtensionWhitelist = new HashSet<string> { ".cs", ".csx", ".vb" };
+        private readonly DiagnosticDescriptor rule;
 
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
         protected bool IsAnalyzerEnabled { get; set; }
         protected bool IgnoreHeaderComments { get; set; }
         protected virtual bool AnalyzeGeneratedCode { get; set; }
+        protected virtual bool AnalyzeTestProjects => true;
         protected string OutPath { get; set; }
         protected bool IsTestProject { get; set; }
 
+        protected UtilityAnalyzerBase(string diagnosticId, string title) =>
```

Comment:
The diagnosticId and title could be protected virtual properties, and a bunch of constructor calls could be removed.

Reply:
I prefer constructor to construct the class in this case :) It syntactically enforces everyone to do it correctly in single line manner.

Reply:
..and we don't need to store nor access them afterwards from the class and it's inheritors so we don't need properties.